projects/online/online/utils/dataloading.py

```diff
@@ -0,0 +1,257 @@
+import logging
+import re
+import time
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from gwpy.timeseries import TimeSeries
+from scipy.signal import resample
+
+PATH_LIKE = Union[str, Path]
+GWF_SAMPLE_RATE = 16384
+
+patterns = {
+    "prefix": "[a-zA-Z0-9_:-]+",
+    "start": "[0-9]{10}",
+    "duration": "[1-9][0-9]*",
+    "suffix": "(gwf)|(hdf5)|(h5)",
+}
+groups = {k: f"(?P<{k}>{v})" for k, v in patterns.items()}
+pattern = "{prefix}-{start}-{duration}.{suffix}".format(**groups)
+fname_re = re.compile(pattern)
+
+
+def parse_frame_name(fname: PATH_LIKE) -> Tuple[str, int, int]:
+    """Use the name of a frame file to infer its initial timestamp and length
+
+    Expects frame names to follow a standard nomenclature
+    where the name of the frame file ends {prefix}_{timestamp}-{length}.gwf
+
+    Args:
+        fname: The name of the frame file
+    Returns:
+        The prefix of the frame file name
+        The initial GPS timestamp of the frame file
+        The length of the frame file in seconds
+    """
+
+    if isinstance(fname, Path):
+        fname = fname.name
+
+    match = fname_re.search(fname)
+    if match is None:
+        raise ValueError(f"Could not parse frame filename {fname}")
+
+    prefix, start, duration, *_ = match.groups()
+    return prefix, int(start), int(duration)
+
+
+def _is_gwf(match):
+    return match is not None and match.group("suffix") == "gwf"
+
+
+def get_prefix(datadir: Path):
+    if not datadir.exists():
+        raise FileNotFoundError(f"No data directory '{datadir}'")
+
+    fnames = map(str, datadir.iterdir())
+    matches = map(fname_re.search, fnames)
+    matches = list(filter(_is_gwf, matches))
+
+    if len(matches) == 0:
+        raise ValueError(f"No valid .gwf files in data directory '{datadir}'")
+
+    t0 = max([int(i.group("start")) for i in matches])
+    prefixes = set([i.group("prefix") for i in matches])
+    if len(prefixes) > 1:
+        raise ValueError(
+            "Too many prefixes {} in data directory '{}'".format(
+                list(prefixes), datadir
+            )
+        )
+
+    durations = set([i.group("duration") for i in matches])
+    if len(durations) > 1:
+        raise ValueError(
+            "Too many lengths {} in data directory '{}'".format(
+                list(durations), datadir
+            )
+        )
+    return list(prefixes)[0], int(list(durations)[0]), t0
+
+
+def reset_t0(datadir, last_t0):
+    tick = time.time()
+    while True:
+        matches = [fname_re.search(i.name) for i in datadir.iterdir()]
+        t0s = np.array([int(i.group("start")) for i in matches if _is_gwf(i)])
+        # t0s = t0s[t0s > last_t0]
```

Comment:
Can delete this, it was used when we didn't reset to the most recent time

Reply:
@wbenoit26 Can we remove this commented line?

Reply:
Yup

Reply:
And thus also `last_t0` as an argument to `reset_t0`?

Reply:
Or do we wan't to just keep that for logging?

Reply:
I'm ambivalent on that one. No harm done in keeping it, but that logging message doesn't ever really come up